### ISSUE TITLE
Outdated deps preventing build

### ISSUE THREAD

**Issue by Bob-The-Marauder** [martinsumner/leveled#384]
The version of config.guess on snappy-1.0.4 is from June 2009 which can cause the following issue:

```
checking whether make sets $(MAKE)... yes
checking build system type... ./config.guess: unable to guess system type

This script, last modified 2009-06-10, has failed to recognize
the operating system you are using. It is advised that you
download the most up to date version of the config scripts from

  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD
and
  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD

If the version you run (./config.guess) is already up to date, please
send the following data and any information you think might be
pertinent to <config-patches@gnu.org> in order to provide the needed
information to handle your system.

config.guess timestamp = 2009-06-10

uname -m = aarch64
uname -r = 5.15.61-v8+
uname -s = Linux
uname -v = #1579 SMP PREEMPT Fri Aug 26 11:16:44 BST 2022

/usr/bin/uname -p = unknown
/bin/uname -X     =

hostinfo               =
/bin/universe          =
/usr/bin/arch -k       =
/bin/arch              = aarch64
/usr/bin/oslevel       =
/usr/convex/getsysinfo =

UNAME_MACHINE = aarch64
UNAME_RELEASE = 5.15.61-v8+
UNAME_SYSTEM  = Linux
UNAME_VERSION = #1579 SMP PREEMPT Fri Aug 26 11:16:44 BST 2022
configure: error: cannot guess build type; you must specify one
===> Hook for compile failed!

make[1]: *** [Makefile:29: compile] Error 1
```

Would it be possible to update to a newer version of either Snappy or at least the config.guess held within to prevent this from happenening, please? I suppose a manual way to specify version might also be a way around this.

**Comment by martinsumner** [comment]
I'm a bit confused. Did you mean to raise an issue on leveldb?  leveled uses lz4 or native (zlib) compression, and doesn't fetch snappy.

Leveldb has had snappy updated to 1.1.9 already though - https://github.com/basho/eleveldb/pull/266

**Comment by Bob-The-Marauder** [comment]
I should apoligise. I got confused between eleveldb and leveled.

I got the above error when building Riak KV 3.0.11 on aarch64 architecture. From searching for `config.guess` in the build files, the only one that popped up was under snappy found in  c_src for the eleveldb directory. There were other occurrences of `config.guess` in the OTP 22.3 build files but I doubt that an OTP build from March 2020 would use a version of config.guess from 2009. Given that the version of snappy used was 1.0.4, it seemed reasonable to assume that was the culprit.

Here's the full error:
```
===> Compiling eleveldb
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking build system type... ./config.guess: unable to guess system type

This script, last modified 2009-06-10, has failed to recognize
the operating system you are using. It is advised that you
download the most up to date version of the config scripts from

  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD
and
  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD

If the version you run (./config.guess) is already up to date, please
send the following data and any information you think might be
pertinent to <config-patches@gnu.org> in order to provide the needed
information to handle your system.

config.guess timestamp = 2009-06-10

uname -m = aarch64
uname -r = 5.15.61-v8+
uname -s = Linux
uname -v = #1579 SMP PREEMPT Fri Aug 26 11:16:44 BST 2022

/usr/bin/uname -p = unknown
/bin/uname -X     = 

hostinfo               = 
/bin/universe          = 
/usr/bin/arch -k       = 
/bin/arch              = aarch64
/usr/bin/oslevel       = 
/usr/convex/getsysinfo = 

UNAME_MACHINE = aarch64
UNAME_RELEASE = 5.15.61-v8+
UNAME_SYSTEM  = Linux
UNAME_VERSION = #1579 SMP PREEMPT Fri Aug 26 11:16:44 BST 2022
configure: error: cannot guess build type; you must specify one
===> Hook for compile failed!

make[1]: *** [Makefile:29: compile] Error 1
make[1]: Leaving directory '/home/pi/riak/rel/pkg/out/riak-3.0.11-OTP22.3'
make: *** [debian/rules:20: build] Error 2
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
debuild: fatal error at line 1182:
dpkg-buildpackage -us -uc -ui -i -b failed
make[2]: *** [/home/pi/riak/rel/pkg/deb/Makefile:10: default] Error 29
make[2]: Leaving directory '/home/pi/riak/rel/pkg/out/riak-riak-3.0.11-OTP22.3'
make[1]: *** [rel/pkg/Makefile:86: ostype] Error 2
make[1]: Leaving directory '/home/pi/riak'
make: *** [Makefile:236: package] Error 2
```

and then when I look for `config.guess`:

```
pi@TIOT-RP4-017:~/riak $ locate config.guess
/home/pi/.kerl/builds/22.3/otp_src_22.3/erts/autoconf/config.guess
/home/pi/.kerl/builds/22.3/otp_src_22.3/lib/common_test/priv/auxdir/config.guess
/home/pi/.kerl/builds/22.3/otp_src_22.3/lib/common_test/test_server/config.guess
/home/pi/.kerl/builds/22.3/otp_src_22.3/lib/erl_interface/src/auxdir/config.guess
/home/pi/.kerl/builds/22.3/otp_src_22.3/lib/wx/autoconf/config.guess
/home/pi/riak/rel/pkg/out/riak-3.0.11-OTP22.3/_build/default/lib/eleveldb/c_src/snappy-1.0.4/config.guess
/usr/share/automake-1.16/config.guess
/usr/share/libtool/build-aux/config.guess
/usr/share/man/man1/config.guess.1.gz
/usr/share/misc/config.guess
```

As the only one referencing Riak is `/home/pi/riak/rel/pkg/out/riak-3.0.11-OTP22.3/_build/default/lib/eleveldb/c_src/snappy-1.0.4/config.guess` and the fact that the error is thrown when building eleveldb, it makes me inclined to think that this is the culprit.

I saw your comment above saying that leveldb is using snappy 1.1.9 so I am puzzled as to why this is using snappy 1.0.4 still in the most recent build tag of KV 3.

I admit that this issue has clearly been placed in the wrong repo. Should I repost it to eleveldb or is it ok to leave it here and apologise for not being able to distinguish between leveled and eleveldb?

**Comment by martinsumner** [comment]
If you could reopen in leveldb, it will be neater.

Sorry, my memory is failing as well.  The fix to move to 1.1.9 of snappy was merged into `develop` not `develop-3.0`, as it was there for the 3.2 release (but not 3.0).

It might be that it can also be made to `develop-3.0`, though it will bring the change of needing cmake to build riak to 3.0 as well.

As @hmmr did this change, it would be best to involve Andre in the discussion about how hard/easy this and any potential issues.
